tests/unit/utils/sparqlhandler/test_csvw_inspector.py

```diff
@@ -0,0 +1,193 @@
+
+    with pytest.raises(KeyError) as exception:
+        assert csvw_inspector.get_primary_catalog_metadata()
+
+    assert (
+        f"Could not find catalog metadata in primary graph '{csvw_inspector.primary_graph_uri}'."
+        in str(exception.value)
+    )
+
+
+def test_detect_csvw_type_qb_dataset():
+    """
+    Tests the detection of the csvw_type of an input metadata json file that has a CsvWRdfManager
+    object created from it, in this case to correctly detect a QbDataset csvw type from its csvw_inspector.
+    """
+    csvw_metadata_json_path = (
+        _test_case_base_dir
+        / "pivoted-single-measure-dataset"
+        / "qb-id-10004.csv-metadata.json"
+    )
+    csvw_rdf_manager = get_csvw_rdf_manager(csvw_metadata_json_path)
+
+    csvw_type = csvw_rdf_manager.csvw_inspector.csvw_type
+    assert csvw_type == CSVWType.QbDataSet
+
+
+def test_csvw_type_key_error():
+    """
+    Test to ensure that when a conjunctive graph that is not recognised by the
+    csvw_type() function as a code list or a qb data set returns a type error.
+    """
+
+    rdf_graph = ConjunctiveGraph()
+
+    csvw_inspector = CsvWInspector(rdf_graph, Path("does-not-exist.csv-metadata.json"))
+
+    with pytest.raises(TypeError) as exception:
+        assert csvw_inspector.csvw_type()
+
+    assert (
+        "The input metadata is invalid as it is not a data cube or a code list."
+        in str(exception.value)
+    )
+
+
+def test_detect_csvw_type_code_list():
+    """
+    Tests the detection of the csvw_type of an input metadata json file that has a CsvWRdfManager
+    object created from it, in this case to correctly detect a CodeList csvw type from its csvw_inspector.
+    """
+    csvw_metadata_json_path = (
+        _test_case_base_dir
+        / "pivoted-single-measure-dataset"
+        / "some-dimension.csv-metadata.json"
+    )
+    csvw_rdf_manager = get_csvw_rdf_manager(csvw_metadata_json_path)
+
+    csvw_type = csvw_rdf_manager.csvw_inspector.csvw_type
+    assert csvw_type == CSVWType.CodeList
+
+
+def test_get_table_info_for_csv_url():
+    """
+    Ensures that the correct table schema properties are returned for the given code list.
+    """
+    csvw_metadata_json_path = (
+        _test_case_base_dir
+        / "pivoted-single-measure-dataset"
+        / "some-dimension.csv-metadata.json"
+    )
+    csvw_rdf_manager = get_csvw_rdf_manager(csvw_metadata_json_path)
+
+    csvw_inspector: CsvWInspector = CsvWInspector(
+        csvw_rdf_manager.rdf_graph, csvw_metadata_json_path
+    )
+
+    result = csvw_inspector.get_table_info_for_csv_url("some-dimension.csv")
+
+    assert result.about_url == "some-dimension.csv#{+uri_identifier}"
+    assert result.csv_url == "some-dimension.csv"
+    assert result.primary_key_col_names == ["uri_identifier"]
+
+
+def test_get_table_info_multiple_primary_keys():
+    """
+    Ensures that primary keys can be retrieved successfully from a table that contains multiple primary keys.
```

Comment:
This shouldn't specifically be about primary keys.

And could you please remove the bit of the comment which talks about the ordering of the primary keys since we realised it wasn't possible to extract them with the order maintained from the RDF graph.

Reply:
We decided to remove this unit test because we realized the new test (below it) now covers everything this test did, but also asserts a few extra things, so we think this test is now redundant. Though we can put it back and change it if that is best.